core/src/main/java/com/hcl/ecomm/core/services/impl/WishListServiceimpl.java

```diff
@@ -0,0 +1,167 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(WishListServiceimpl.class);
+
+    @Reference
+    LoginService loginService;
+
+    @Activate
+    private MagentoServiceConfig config;
+
+    @Override
+    public String getDomainName() {
+        return loginService.getDomainName();
+    }
+
+
+    @Override
+    public String getAddToWishListPath() {
+        return config.customer_addToWishList_string();
+    }
+
+    @Override
+    public String getWishListItemsPath() {
+        return config.customer_getWishList_string();
+    }
+
+    private String responseStream = null;
+    private String schema = "http";
+    @Override
+    public JSONObject addToWishList(String Sku, String custToken) {
+        
+        LOG.debug("addToWishList method start  product={}: " + Sku);
+
+        String addTowishlistPath = "";
+        String authToken = "";
+        JSONObject addToWishlistResponse = new JSONObject();;
+
+        try {
+            if(custToken != null && !custToken.isEmpty()) {
+                authToken = custToken;
+                addTowishlistPath = getAddToWishListPath();
+            }
+            String domainName = getDomainName();
+           addTowishlistPath = addTowishlistPath.replace("sku", Sku);
+            String url = schema + "://" + domainName + addTowishlistPath;
+            LOG.info("addTowishlistPath : " + url);
+
+            Integer statusCode;
+
+
+            CloseableHttpClient httpClient = HttpClients.createDefault();
+            HttpPost httppost = new HttpPost(url);
+            httppost.setHeader("Content-Type", "application/json");
+            httppost.setHeader("Authorization", "Bearer " +authToken);
+            CloseableHttpResponse httpResponse = httpClient.execute(httppost);
+            statusCode = httpResponse.getStatusLine().getStatusCode();
+            LOG.info("add to wishlist: magento statusCode ={}",statusCode);
+            JSONObject wishlistresponse = new JSONObject();
+            if(HttpStatus.SC_OK == statusCode){
+                BufferedReader br = new BufferedReader(new InputStreamReader((httpResponse.getEntity().getContent())));
+                String str="";
+                String output;
+                while ((output = br.readLine()) != null) {
+
+                    str +=output;
+                }
+
+                wishlistresponse.put("response",str);
+                addToWishlistResponse.put("statusCode", statusCode);
+                addToWishlistResponse.put("message", wishlistresponse);
+            }else if(HttpStatus.SC_BAD_REQUEST == statusCode){
+                BufferedReader br = new BufferedReader(new InputStreamReader((httpResponse.getEntity().getContent())));
+                String str="";
+                String output;
+                while ((output = br.readLine()) != null) {
+
+                    str +=output;
+                }
+
+                wishlistresponse.put("response",str);
```

Comment:
Can we remove buffered reader and corresponding loop for else if block for bad request .Instead of that we can add http Response to message String

Reply:
updated as per your comment